src/Middleware/HeaderPropagation/test/HeaderPropagationMessageHandlerTest.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Xunit;
+
+namespace Microsoft.AspNetCore.HeaderPropagation.Tests
+{
+    public class HeaderPropagationMessageHandlerTest
+    {
+        public HeaderPropagationMessageHandlerTest()
+        {
+            Handler = new SimpleHandler();
+
+            State = new HeaderPropagationState();
+            Options = new HeaderPropagationOptions();
+
+            var headerPropagationMessageHandler =
+                new HeaderPropagationMessageHandler(new OptionsWrapper<HeaderPropagationOptions>(Options), State)
```

Comment:
You can also use `Options.Create`, it's a little less verbose`

Reply:
ok